ast.go

```diff
@@ -875,6 +875,42 @@ func (node ColumnList) String() string {
 	return fmt.Sprintf(" %s%s%s", "(", strings.Join(strs, ", "), ")")
 }
 
+// IndexedColumn represents a indexed column.
```

Comment:
While doing this PR, I noticed that the [table constraint](https://www.sqlite.org/syntax/table-constraint.html) implementation was incomplete. The `PRIMARY KEY` constraint what accepting only a column name list (see [grammar diff](https://github.com/tablelandnetwork/go-sqlparser/pull/29/files#diff-a6ec253416e8e9e6846f73f5c7677d6ff6fa85f5c37317f84e608fc50ff8238eL1173-R1228)). But it needs to accept a more robust structure called [indexed column](https://www.sqlite.org/syntax/indexed-column.html). The `IndexedColum` node represents that structure.

Reply:
I represented `IndexedColumn` as a `Column` with `CollationName` and `Order` although the syntax diagram says it can have an arbitrary `expr`.
![Screen Shot 2022-11-16 at 18 20 46](https://user-images.githubusercontent.com/1233473/202296981-af7af45d-dd22-485b-b8f0-e6ba39d116ac.png)

I excluded the `expr` representation in our AST because I could not figure out an `expr` that SQLite accepts as valid. For example, the following statements are not allowed (i discovered the below by trying it myself):

```
CREATE TABLE a (a int, PRIMARY KEY ( 0  DESC))
CREATE TABLE a (a int, PRIMARY KEY (a > 0  DESC))
CREATE TABLE a (a int, PRIMARY KEY ((SELECT 1) DESC))
```

So there's no point in allowing an arbitrary `expr`. A column is all we need.

The `CollationName` can be an empty identifier and `Order` an empty string to indicate the optionality.

Reply:
Sounds good, but I think only accepting column names was fine too.